wsclient/src/main/java/org/hpccsystems/ws/client/platform/WorkunitInfo.java

```diff
@@ -69,27 +74,36 @@
     private String sourceFileDesc;
     private Integer timerCount;
     private String timersDesc;
-    private ECLTimingData[] timingData;
+    private List<ECLTimingDataWrapper> timingData=new ArrayList<ECLTimingDataWrapper>();
     private String totalClusterTime;
     private Integer variableCount;
     private List<ECLResultInfo> variables;
     private boolean thorLCR;
-    private ThorLogInfo[] thorLogList;
+    private List<ThorLogInfoWrapper> thorLogList=new ArrayList<ThorLogInfoWrapper>();
     private String variablesDesc;
     private Integer warningCount;
     private Integer workflowCount;
-    private ECLWorkflow[] workflows;
+    private List<ECLWorkflowWrapper> workflows=new ArrayList<ECLWorkflowWrapper>();
     private String workflowsDesc;
     private String wuid;
     private String xmlParams;
-    private ECLWorkunit eclWorkunit;
     private String[] resultViews;
     /**
      * Create an ECL workunit from a axis-generated soap class ECL Workunit
      * 
      * @param base
      */
-    public WorkunitInfo(ECLWorkunit base)
+    public WorkunitInfo(org.hpccsystems.ws.client.gen.wsworkunits.v1_73.ECLWorkunit base)
```

Comment:
it feels like all wrappers, but especially workunitInfo should have a private field that indicates the runtime version it is representing, and it should make decisions based on that version

Reply:
Good idea that. But the class itself will still need multiple constructors and multiple getRaws() to handle being used against multiple clusters